velox/type/fbhive/HiveTypeParser.cpp

```diff
@@ -87,7 +88,16 @@ std::shared_ptr<const Type> HiveTypeParser::parse(const std::string& ser) {
 Result HiveTypeParser::parseType() {
   Token nt = nextToken();
   VELOX_CHECK(!nt.isEOS(), "Unexpected end of stream parsing type!!!");
-  if (nt.isValidType() && nt.isPrimitiveType()) {
+
+  if (!nt.isValidType()) {
+    VELOX_FAIL(fmt::format(
```

Comment:
I know you're only copying it, but you can omit the fmt::format as Velox exception macros can do the formatting automatically for you.